src/components/InnerBlockSlider/index.js

```diff
@@ -25,12 +25,23 @@ const InnerBlockSlider = ( {
 } ) => {
 	const innerBlockTemplate = template || [ [ allowedBlock ] ];
 
-	const slideBlocks = useSelect(
-		( select ) =>
-			select( 'core/block-editor' ).getBlock( parentBlockId ).innerBlocks
+	const {
+		slideBlocks,
+		selectedBlockId,
+		hasSelectedInnerBlock,
+	 } = useSelect(
+		( select ) => {
+			const blockEditorStore = select( 'core/block-editor' );
```

Comment:
Maybe you can use `getBlockParentsByBlockName` to find the parent slide from any level of nested children https://developer.wordpress.org/block-editor/reference-guides/data/data-core-block-editor/#getblockparentsbyblockname

Reply:
Neat!

Reply:
Actually this didn't quite work. But I ended up using `getLowestCommonAncestorWithSelectedBlock` which handled deeper nesting.

Reply:
Ah ok, good to know

Reply:
Noting that it didn't work because it `getBlockParentsByBlockName` only returned direct parents, and I wanted to check all ancestors